clients/client-java/src/main/java/org/apache/gravitino/client/GravitinoMetalake.java

```diff
@@ -1189,6 +1204,162 @@ public String[] listBindingRoleNames() {
     return metadataObjectRoleOperations.listBindingRoleNames();
   }
 
+  @Override
+  public List<JobTemplate> listJobTemplates() {
+    JobTemplateListResponse resp =
+        restClient.get(
+            String.format(API_METALAKES_JOB_TEMPLATES_PATH, RESTUtils.encodeString(this.name())),
+            ImmutableMap.of("details", "true"),
+            JobTemplateListResponse.class,
+            Collections.emptyMap(),
+            ErrorHandlers.jobErrorHandler());
+    resp.validate();
+
+    return resp.getJobTemplates().stream()
+        .map(org.apache.gravitino.dto.util.DTOConverters::fromDTO)
+        .collect(Collectors.toList());
+  }
+
+  @Override
+  public void registerJobTemplate(JobTemplate jobTemplate)
+      throws JobTemplateAlreadyExistsException {
+    JobTemplateRegisterRequest req =
+        new JobTemplateRegisterRequest(DTOConverters.toJobTemplateDTO(jobTemplate));
+    req.validate();
```

Comment:
Should the request be validated by the server instead of the client?

Reply:
This will also be validated in the server side. To vaildate in the client side is to avoid unncessary remote call.

Reply:
There is a case: after users upgrade the server to change certain request limitations, if we validate the request on the client side, they will also need to upgrade the client.

Reply:
I see, I will remove it.